### PR TITLE
test: Migrate Frontend tests to a GHA Workflow

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**
Resolves https://github.com/kubeflow/pipelines/issues/10879

Converts the Prow frontend test to a GH Action Workflow.

Here's a working GH Action run on my fork: https://github.com/DharmitD/data-science-pipelines-argo/actions/runs/9525123655/job/26258863870

PR to remove the frontend test from prow config: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2302

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by gmfrasca — COMMENTED** [review]
GHaction /lgtm , but I think we need to remove the `kubeflow-pipeline-frontend-test` from the CI checks as this supercedes the test and removes the prow ci check, but is still marked as required

**Comment by DharmitD** [comment]
> GHaction /lgtm , but I think we need to remove the `kubeflow-pipeline-frontend-test` from the CI checks as this supercedes the test and removes the prow ci check, but is still marked as required

Find PR to remove the frontend test from prow config here: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2302

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    paths:
+      - 'frontend/**'
```
Comment (hbelmiro) [inline .github/workflows/frontend.yml:11]: I think we should also add this workflow itself.

```suggestion
      - 'frontend/**'
      - '.github/workflows/frontend.yml'
```

File: frontend/package.json [hunk 2]
```diff
@@ -92,7 +92,6 @@
     "test:server:coverage": "cd ./server && npm test -- --coverage && cd ..",
     "test:coverage": "npm test -- --coverage && npm run test:server:coverage",
     "test:ci": "export CI=true && npm run format:check && npm run lint && npm run test:coverage",
-    "test:ci:prow": "npm set unsafe-perm true && npm ci && npm run test:ci && ./scripts/report-coveralls.sh",
```
Comment (hbelmiro) [inline frontend/package.json:95]: We can also remove the `./scripts/report-coveralls.sh` and `frontend/scripts/get-coveralls-repo-token.js` files. Seems like they are no longer referenced with this change.

File: .github/workflows/frontend.yml (hunk 1 above)
Comment (DharmitD) [reply .github/workflows/frontend.yml:11]: done, added this workflow to paths.

File: frontend/package.json (hunk 2 above)
Comment (DharmitD) [reply frontend/package.json:95]: Done, removed these files.

**Comment by DharmitD** [comment]
This PR is ready for review.
cc: @hbelmiro @rimolive @chensun

**Review by hbelmiro — COMMENTED** [review]
/lgtm

**Comment by rimolive** [comment]
/lgtm

@chensun @zijianjoy

**Comment by DharmitD** [comment]
@chensun could you please approve?

**Comment by DharmitD** [comment]
cc @chensun @zijianjoy @james-jwu

**Review by chensun — APPROVED** [review]
/lgtm

**Comment by DharmitD** [comment]
@chensun could you also approve https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2302?
And merge both these PRs? Thank you!

cc: @zijianjoy @james-jwu

**Comment by DharmitD** [comment]
Closing this PR and re-opening so that the tide check runs using latest config.

**Comment by DharmitD** [comment]
Reopening.

**Comment by DharmitD** [comment]
Duplicated here: https://github.com/kubeflow/pipelines/pull/10972

**Pull request closed without merging**